packages/oauth-encryption/encrypt.js

```diff
@@ -96,23 +96,24 @@ OAuthEncryption.open = function (ciphertext, userId) {
       throw new Error();
     }
 
-    var result = gcm.decrypt(
+    var decipher = crypto.createDecipheriv(
+      "aes-128-gcm",
       gcmKey,
-      new Buffer(ciphertext.iv, "base64"),
-      new Buffer(ciphertext.ciphertext, "base64"),
-      new Buffer([]), /* aad */
-      new Buffer(ciphertext.authTag, "base64")
+      new Buffer(ciphertext.iv, "base64")
     );
 
-    if (! result.auth_ok) {
-      throw new Error();
-    }
+    decipher.setAAD(new Buffer([]));
+    decipher.setAuthTag(new Buffer(ciphertext.authTag, "base64"));
+    var chunks = [decipher.update(
+      new Buffer(ciphertext.ciphertext, "base64"))];
+    chunks.push(decipher.final());
```

Comment:
Not clear to me at which step crypto throws or otherwise indicates decryption or authentication errors but it looks like this change does affect how errors are detected? Or do you need to add an error handler?